pkg/builtin/multi/ceil.go

```diff
@@ -0,0 +1,368 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multi
+
+import (
+	"errors"
+	"fmt"
+	"github.com/matrixorigin/matrixone/pkg/vectorize/ceil"
+
+	"github.com/matrixorigin/matrixone/pkg/builtin"
+	"github.com/matrixorigin/matrixone/pkg/container/nulls"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/encoding"
+	"github.com/matrixorigin/matrixone/pkg/sql/colexec/extend"
+	"github.com/matrixorigin/matrixone/pkg/sql/colexec/extend/overload"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+)
+
+func init() {
+	extend.FunctionRegistry["ceil"] = builtin.Ceil
+	for _, item := range argsAndRets {
+		overload.AppendFunctionRets(builtin.Ceil, item.args, item.ret)
+	}
+	extend.MultiReturnTypes[builtin.Ceil] = func(es []extend.Extend) types.T {
+		return getMultiReturnType(builtin.Ceil, es)
+	}
+
+	extend.MultiStrings[builtin.Ceil] = func(es []extend.Extend) string {
+		if len(es) > 1 {
+			return fmt.Sprintf("ceil(%s, %s)", es[0], es[1])
+		} else {
+			return fmt.Sprintf("ceil(%s)", es[0])
+		}
+	}
+	overload.OpTypes[builtin.Ceil] = overload.Multi
+	overload.MultiOps[builtin.Ceil] = []*overload.MultiOp{
+		{
+			Min:        1,
+			Max:        2,
+			Typ:        types.T_uint8,
+			ReturnType: types.T_uint8,
+			Fn: func(vecs []*vector.Vector, proc *process.Process, cs []bool) (*vector.Vector, error) {
+				digits := int64(0)
+				vs := vecs[0].Col.([]uint8)
+				if len(vecs) > 1 {
+					if !cs[1] || vecs[1].Typ.Oid != types.T_int64 {
+						return nil, errors.New("The second argument of the ceil function must be an int64 constant")
+					}
+					digits = vecs[1].Col.([]int64)[0]
+				}
+				if vecs[0].Ref == 1 || vecs[0].Ref == 0 {
+					vecs[0].Ref = 0
+					ceil.CeilUint8(vs, vs, digits)
+					return vecs[0], nil
+				}
+				vec, err := process.Get(proc, int64(len(vs)), types.Type{Oid: types.T_uint8, Size: 1})
+				if err != nil {
+					return nil, err
+				}
+				rs := encoding.DecodeUint8Slice(vec.Data)
+				rs = rs[:len(vs)]
+				vec.Col = rs
+				nulls.Set(vec.Nsp, vecs[0].Nsp)
+				vector.SetCol(vec, ceil.CeilUint8(vs, rs, digits))
+				return vec, nil
+			},
+		},
+		{
+			Min:        1,
+			Max:        2,
+			Typ:        types.T_uint16,
+			ReturnType: types.T_uint16,
+			Fn: func(vecs []*vector.Vector, proc *process.Process, cs []bool) (*vector.Vector, error) {
+				digits := int64(0)
+				vs := vecs[0].Col.([]uint16)
+				if len(vecs) > 1 {
+					if !cs[1] && vecs[1].Typ.Oid != types.T_int64 {
```

Comment:
perhaps we need to change here too, the second argument is not a constant **or** the second argument type is not int64..., I think the other lines below have the same mistake

Reply:
Sorry, I have made changes, the server of the database cannot be connected on my local machine, so I cannot debug effectively

Reply:
actually, in matrixone, the constant integer passed in from the frontend will be casted to a int64, so I think the original check was correct
```go
vecs[1].Typ.Oid != types.T_int64
```
what I believe is incorrect here is the logical operation && here, we should return error if any checks here are not satisfied, hence || 
```go
!cs[1] || vecs[1].Typ.Oid != types.T_int64
```

Reply:
> Sorry, I have made changes, the server of the database cannot be connected on my local machine, so I cannot debug effectively

you can connect to MatrixOne via a mysql client, after setting up mo-server, run this command in the shell:
```
mysql -h 127.0.0.1 -P 6001 -udump -p
```
or use docker, the project README has the details